### DISCUSSION TITLE
Get attributes based on edge id

### DISCUSSION THREAD

**Discussion by w158937299** [JanusGraph/janusgraph#4522] [Q&A] [answered]
It is known that the id of an edge is {'outVertexId': 4120, 'typeId': 10261, 'relationId': 27651, 'inVertexId': 20544}, so how do we get the corresponding attribute based on this id, I have tried many ways but I am unable to get it. Could you please provide me with the appropriate query?

**Comment by li-boxuan** [comment, marked as answer]
`g.V(4120).outE().where(inV().hasId(20544)).values('prop')` if you know there's only one edge between these two vertices.

You can also use RelationIdentifier::toString method to construct the edge out of this representation. E.g.

```java
public class App {
    public static void main(String[] args) {
        org.janusgraph.graphdb.relations.RelationIdentifier id = new RelationIdentifier(4120, 10261, 27651, 20544);
        System.out.println(id);
    }
}
```

which would print out the result: `lc3-36g-7x1-fuo`.

So you could use `g.E("lc3-36g-7x1-fuo")` to find this edge.

**Reply by w158937299** [reply]
Thanks for your answer, I was also wondering if I could get an id like 'lc3-36g-7x1-fuo' with a gremlin query.

**Reply by li-boxuan** [reply]
> with a gremlin query

Not possible at the moment, but we could probably add that support. If'd like to, create a feature request by opening a GitHub issue.
